articles/flask-crud-api/index.md

```diff
@@ -0,0 +1,306 @@
+
+    def create(self):
+        db.session.add(self)
+        db.session.commit()
+        return self
+
+    def __init__(self, title, todo_description):
+        self.title = title
+        self.todo_description = todo_description
+
+    def __repr__(self):
+        return f"{self.id}"
+
+
+db.create_all()
+
+
+class TodoSchema(ModelSchema):
+    class Meta(ModelSchema.Meta):
+        model = Todo
+        sqla_session = db.session
+
+    id = fields.Number(dump_only=True)
+    title = fields.String(required=True)
+    todo_description = fields.String(required=True)
+
+
+@app.route('/api/v1/todo', methods=['GET'])
+def index():
+    get_todos = Todo.query.all()
+    todo_schema = TodoSchema(many=True)
+    todos = todo_schema.dump(get_todos)
+    return make_response(jsonify({"todos": todos}))
+
+
+@app.route('/api/v1/todo/<id>', methods=['GET'])
+def get_todo_by_id(id):
+    get_todo = Todo.query.get(id)
+    todo_schema = TodoSchema()
+    todo = todo_schema.dump(get_todo)
+    return make_response(jsonify({"todo": todo}))
+
+
+@app.route('/api/v1/todo/<id>', methods=['PUT'])
+def update_todo_by_id(id):
+    data = request.get_json()
+    get_todo = Todo.query.get(id)
+    if data.get('title'):
+        get_todo.title = data['title']
+    if data.get('todo_description'):
+        get_todo.todo_description = data['todo_description']
+    db.session.add(get_todo)
+    db.session.commit()
+    todo_schema = TodoSchema(only=['id', 'title', 'todo_description'])
+    todo = todo_schema.dump(get_todo)
+    return make_response(jsonify({"todo": todo}))
+
+
+@app.route('/api/v1/todo/<id>', methods=['DELETE'])
+def delete_todo_by_id(id):
+    get_todo = Todo.query.get(id)
+    db.session.delete(get_todo)
+    db.session.commit()
+    return make_response("", 204)
+
+
+@app.route('/api/v1/todo', methods=['POST'])
+def create_todo():
+    data = request.get_json()
+    todo_schema = TodoSchema()
+    todo = todo_schema.load(data)
+    result = todo_schema.dump(todo.create())
+    return make_response(jsonify({"todo": result}), 200)
+
+
+if __name__ == "__main__":
+    app.run(debug=True)
+
+```
+To run our application, execute the command below in the command line.
+```bash
+$ flask run
+```
+### Testing the API endpoints
+**GET endpoint**
+
+![Get all todos](/engineering-education/flask-crud-api/get-todos.png)
+
+![Get todo by id](/engineering-education/flask-crud-api/get-todo-by-id.png)
+
+**POST endpoint**
+
+![Create todo](/engineering-education/flask-crud-api/create-todo.png)
+
+**PUT endpoint**
+
+![Update todo](/engineering-education/flask-crud-api/update-todo.png)
+
+**DELETE endpoint**
+
+![Delete todo](/engineering-education/flask-crud-api/delete-todo.png)
+
+### Conclusion
+Now that you have learned how to create a CRUD API using flask, add the `date_created` and `last_modified` fields to our `Todo` model and make modifications to various API endpoints to include the added fields in the API requests and responses.
+
```

Comment:
This sentence is too long. Consider breaking it down. Also, it would be important to talk about the [datetime](https://docs.python.org/3/library/datetime.html) python module here.